src/java/main/org/apache/zookeeper/client/StaticHostProvider.java

```diff
@@ -45,9 +46,12 @@
 
     private int currentIndex = -1;
 
+    // Don't re-resolve on first next() call
+    private boolean connectedSinceNext = true;
```

Comment:
Should this be initialised to `true`? Alternatively we could change the name of the variable.

Reply:
It's initialized to true to make sure that on the first pass through next, you don't try to re-resolve.

I cribbed this from the apache PR, I don't know of a better thing to call it.

Reply:
OK. Let's keep this in line with the apache PR then. Just seemed weird to start in a connected state. Would an `Optional` be an option?

Reply:
ZK 3.4 supports JDK 1.6 (I've used some 1.7 things however, as we only use this in 1.8), and the build stack doesn't like Optional IIRC.